syntax = "proto3";

package Course;

// The Course service definition.
service Course {
  rpc GetAllCourse (CourseData) returns (CourseList) {}
  rpc CreateCourse (CourseData) returns (CourseData) {}
  rpc UpdateCourse (CourseData) returns (CourseData) {}
  rpc DeleteCourse (CourseId) returns (CourseId) {}
}

// Message representing a single course record
message CourseData {
  int32 id = 1;
  string name = 2;
  string details = 3;
  int32 industry_id = 4;
  int32 cert_id = 5;
}

// Response message containing a list of courses
message CourseList {
  repeated CourseData courses = 1;
}

message CourseId{
  string courseId = 1;
}

